v2/tools/generator/internal/codegen/embeddedresources/misbehaving_embedded_resources.go

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package embeddedresources
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/config"
+)
+
+type misbehavingResourceCtx struct {
+	resourceName astmodel.TypeName
+	typeName     astmodel.TypeName
+}
+
+func findMisbehavingResources(configuration *config.Configuration, defs astmodel.TypeDefinitionSet) (astmodel.TypeNameSet, error) {
+	resources := make(astmodel.TypeNameSet)
+	visitor := astmodel.TypeVisitorBuilder{
+		VisitObjectType: func(this *astmodel.TypeVisitor, ot *astmodel.ObjectType, ctx interface{}) (astmodel.Type, error) {
+			typedCtx := ctx.(misbehavingResourceCtx)
+			for _, prop := range ot.Properties() {
+				isResourceLifecycleOwnedByParent, _ := configuration.IsResourceLifecycleOwnedByParent(typedCtx.typeName, prop.PropertyName())
```

Comment:
Need to check whether the error returned is a `NotConfiguredError` - there's a helper method for that. If it's any other kind of error, it needs to be returned.

Reply:
Fixed.